### PR TITLE
🚀 Typst export

### PR DESCRIPTION
**Pull request by rowanc1**
Convert markdown files into typst for PDF creation.

- [ ] Need better templating
- [ ] Remove unknown cross-references
- [ ] Tables
- [ ] Render

### CODE REVIEW

**Comment by rowanc1** [comment]
@fwkoch could you take a look at this tomorrow? I mostly want to know if you have any tips for the exports!

File: packages/myst-cli/src/build/typst/single.ts [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+      extraLinkTransformers,
+      simplifyOutputs: true,
+    },
+  );
+  writeBibtexFromCitationRenderers(
+    session,
+    path.join(path.dirname(templateOptions.output), DEFAULT_BIB_FILENAME),
+  );
+
+  const mystTemplate = new MystTemplate(session, {
```
Comment (fwkoch) [inline packages/myst-cli/src/build/typst/single.ts:144]: This template stuff will be easier once there are actually `typst` templates. Right now there's a bunch of copied code that doesn't actually do anything I don't think.

File: packages/myst-cli/src/build/build.ts [hunk 2]
```diff
@@ -26,29 +27,33 @@ export type BuildOpts = {
 };
 
 export function hasAnyExplicitExportFormat(opts: BuildOpts): boolean {
-  const { docx, pdf, tex, xml, md, meca } = opts;
-  return docx || pdf || tex || xml || md || meca || false;
+  const { docx, pdf, tex, typst, xml, md, meca } = opts;
```
Comment (fwkoch) [inline packages/myst-cli/src/build/build.ts:30]: We just need to refactor this boilerplate somehow to reduce repetition. Some way to specify things declaratively only once

File: packages/myst-cli/src/build/utils/collectExportOptions.ts [hunk 3]
```diff
@@ -301,6 +306,18 @@ export async function collectExportOptions(
           )),
         );
       }
+      if (formats.includes(ExportFormats.typst)) {
+        fileExportOptionsList.push(
+          ...(await collectTexExportOptions(
```
Comment (fwkoch) [inline packages/myst-cli/src/build/utils/collectExportOptions.ts:311]: Probably need to reframe this function as something like `collectThingsThatMayOrMayNotBeZippedExportOptions`

**Comment by fwkoch** [comment]
In general, no tips - I think you are right to reuse the `tex` export since this has similar "zip or not" functionality. Beyond that though, I'd just have to dig in a bit and refactor away some of the repetition.

**Comment by rowanc1** [comment]
Goign to break this into a few different PRs!

**Pull request closed without merging**